package-parser/package_parser/commands/get_api/_model.py

```diff
@@ -350,31 +350,90 @@ def to_json(self) -> Any:
         }
 
 
-class RefinedType:
+class Type:
+    def __init__(
+        self,
+        typestring: ParameterAndResultDocstring,
+    ) -> None:
+        self.type: Union[
+            NamedType, EnumType, BoundaryType, UnionType, None
```

Comment:
It would be worthwhile to introduce an [abstract base class](https://docs.python.org/3/library/abc.html) `AbstractType` that `NamedType`, `EnumType`, `BoundaryType` and `UnionType` inherit from. This makes it possible to add new subclasses of `AbstractType` later without having to update the type hints of all functions that use this.